core/src/main/java/de/danoeh/antennapod/core/syndication/namespace/NSMedia.java

```diff
@@ -23,22 +25,36 @@
 	private static final String SIZE = "fileSize";
 	private static final String MIME_TYPE = "type";
 	private static final String DURATION = "duration";
+	private static final String DEFAULT = "isDefault";
+
+	private static final String IMAGE = "thumbnail";
+	private static final String IMAGE_URL = "url";
+
+	private static final String DESCRIPTION = "description";
+	private static final String DESCRIPTION_TYPE = "type";
 
 	@Override
 	public SyndElement handleElementStart(String localName, HandlerState state,
-			Attributes attributes) {
+										  Attributes attributes) {
 		if (CONTENT.equals(localName)) {
 			String url = attributes.getValue(DOWNLOAD_URL);
 			String type = attributes.getValue(MIME_TYPE);
-            boolean validType;
-            if(SyndTypeUtils.enclosureTypeValid(type)) {
-                validType = true;
-            } else {
-                type = SyndTypeUtils.getValidMimeTypeFromUrl(url);
-                validType = type != null;
-            }
-            if (state.getCurrentItem() != null && state.getCurrentItem().getMedia() == null &&
-                url != null && validType) {
+			String defaultStr = attributes.getValue(DEFAULT);
+			boolean validType;
+			boolean isDefault = false;
+			if (SyndTypeUtils.enclosureTypeValid(type)) {
+				validType = true;
+			} else {
+				type = SyndTypeUtils.getValidMimeTypeFromUrl(url);
+				validType = type != null;
+			}
+
+			if (defaultStr == "true")
```

Comment:
In Java, this checks whether defaultStr and the string literal "true" are the same object, not whether their contents match. You should use "true".equals(defaultStr) instead.